virttest/utils_test/libvirt.py

```diff
@@ -1410,6 +1418,39 @@ def cleanup_dest_vm(self, vm, srcuri, desturi):
         # Set connect uri back to local uri
         vm.connect_uri = srcuri
 
+    def check_vm_state(self, vm, state='paused', uri=None):
+        """
+        checks whether state of the vm is as expected
+
+        :param vm: VM Object
+        :param state: expected state of the VM
+        :param uri: connect uri
+
+        :return: True if state of VM is as expected, False otherwise
+        """
+        if not virsh.domain_exists(vm.name, uri=uri):
+            return False
+        vm_state = virsh.domstate(vm.name, uri=uri).stdout.strip()
+        return vm_state.lower() == state.lower()
+
+    def check_migrate_started(self, vm, state='paused', uri=None, timeout=60):
+        """
+        checks whether migration is started of not
+
+        :param vm: VM object
+        :param state: expected VM state in destination host
+        :param uri: connect uri
+        :param timeout: time in seconds to wait for migration to start
+
+        :return: True if migration is started False otherwise
+        """
+        def check_state():
+            try:
+                return self.check_vm_state(vm, state, uri)
+            except Exception:
+                return False
+        return utils_misc.wait_for(check_state, timeout)
```

Comment:
can we just use check_vm_state() directly, rather than wrap one duplicate func?

Reply:
we have to pass few parameters to the check_vm_state(), so we use duplicate func given input to wait_for()